app/controllers/merchant/discounts_controller.rb

```diff
@@ -14,6 +18,17 @@ def create
     end
   end
 
+  def update
+    discount = current_user.merchant.discounts.find(params[:id])
+    if discount.update(discount_params)
+      flash[:success] = "Discount Updated."
+      redirect_to "/merchant"
+    else
+      flash[:error] = discount.errors.full_messages.to_sentence
+      redirect_to "/merchant/discounts/#{discount.id}/edit"
```

Comment:
Redirect instead of render so can display the previous information entered by the user